test/hotspot/jtreg/compiler/lib/ir_framework/driver/TestVMProcess.java

```diff
@@ -91,15 +91,20 @@ public static String getLastTestVMOutput() {
     }
 
     private void prepareTestVMFlags(List<String> additionalFlags, TestFrameworkSocket socket, Class<?> testClass,
-                                    Set<Class<?>> helperClasses, int defaultWarmup) {
+                                    Set<Class<?>> helperClasses, int defaultWarmup, boolean testClassesOnBootClassPath) {
         // Set java.library.path so JNI tests which rely on jtreg nativepath setting work
         cmds.add("-Djava.library.path=" + Utils.TEST_NATIVE_PATH);
         // Need White Box access in test VM.
-        cmds.add("-Xbootclasspath/a:.");
+        String bootClassPath = "-Xbootclasspath/a:.";
+        if (testClassesOnBootClassPath) {
+            // Add test classes themselves to boot classpath. This is required, for example, for IR tests with @Stable.
+            bootClassPath += ":" + Utils.TEST_CLASSES;
+        }
+        cmds.add(bootClassPath);
         cmds.add("-XX:+UnlockDiagnosticVMOptions");
         cmds.add("-XX:+WhiteBoxAPI");
-        // Ignore CompileCommand flags which have an impact on the profiling information.
-        List<String> jtregVMFlags = Arrays.stream(Utils.getTestJavaOpts()).filter(s -> !s.contains("CompileThreshold")).collect(Collectors.toList());
+        // Ignore CompileThreshold and CompileCommand flags which have an impact on the profiling information.
+        List<String> jtregVMFlags = Arrays.stream(Utils.getTestJavaOpts()).filter(s -> !s.contains("CompileThreshold")).toList();
```

Comment:
This hunk seems unnecessary? I would like to have IR Framework patch that is easily backportable and does not contain additional, non-essential hunks :)

Reply:
Fair point, could be done separately, dropped :)